src/Response/MappedResponseCollection.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace PhpTwinfield\Response;
+
+class MappedResponseCollection extends \ArrayObject
+{
+    /**
+     * @param mixed $value
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function append($value): void
+    {
+        if (!($value instanceof IndividualMappedResponse)) {
+            throw new \InvalidArgumentException("Value has to be an object of type " . IndividualMappedResponse::class);
+        }
+        parent::append($value);
+    }
+
+    public function hasSuccessfulResponses(): bool
+    {
+        return $this->countResponses(true) > 0;
+    }
+
+    public function countSuccessfulResponses(): int
+    {
+        return $this->countResponses(true);
+    }
+
+    /**
+     * @return IndividualMappedResponse[]
+     */
+    public function getSuccessfulResponses()
+    {
+        return $this->getResponses(true);
+    }
+
+    public function hasFailedResponses(): bool
+    {
+        return $this->countResponses(false) > 0;
+    }
+
+    public function countFailedResponses(): int
+    {
+        return $this->countResponses(false);
+    }
+
+    /**
+     * @return IndividualMappedResponse[]
+     */
+    public function getFailedResponses()
+    {
+        return $this->getResponses(false);
+    }
+
+    public function assertAllSuccessful()
+    {
+        assert($this->countResponses(false) === 0);
```

Comment:
Not sure, but maybe you should use webmozart's `Assert` here. Lets ask Assert-guru @willemstuursma.